kivy/ext/mpl/backend_kivy.py

```diff
@@ -0,0 +1,386 @@
+    be a do nothing func.  See the GTK backend for an example of how to handle
+    interactive versus batch mode
+    """
+    for manager in Gcf.get_all_fig_managers():
+        # do something to display the GUI
+        manager.show()
+
+
+def new_figure_manager(num, *args, **kwargs):
+    """
+    Create a new figure manager instance
+    """
+    # if a main-level app must be created, this (and
+    # new_figure_manager_given_figure) is the usual place to
+    # do it -- see backend_wx, backend_wxagg and backend_tkagg for
+    # examples.  Not all GUIs require explicit instantiation of a
+    # main-level app (egg backend_gtk, backend_gtkagg) for pylab
+    FigureClass = kwargs.pop('FigureClass', Figure)
+    thisFig = FigureClass(*args, **kwargs)
+    return new_figure_manager_given_figure(num, thisFig)
+
+
+def new_figure_manager_given_figure(num, figure):
+    """
+    Create a new figure manager instance for the given figure.
+    """
+    canvas = FigureCanvasKivy(figure)
+    manager = FigureManagerKivy(canvas, num)
+    return manager
+
+
+class FigureCanvasKivy(FigureCanvasKivyAgg):
+    """
+    The canvas the figure renders into.  Calls the draw and print fig
+    methods, creates the renderers, etc...
+
+    Public attribute
+
+      figure - A Figure instance
+
+    Note GUI templates will want to connect events for button presses,
+    mouse movements and key presses to functions that call the base
+    class methods button_press_event, button_release_event,
+    motion_notify_event, key_press_event, and key_release_event.  See,
+    e.g., backend_gtk.py, backend_wx.py and backend_tkagg.py
+    """
+    
+    def __init__(self, figure, **kwargs):
+        if _debug:
+            print('FigureCanvasKivy: ', figure)
+        
+        self._keyboard = Window.request_keyboard(self._keyboard_closed, self)
+        self._keyboard.bind(on_key_down = self._on_keyboard_down)
+        self._keyboard.bind(on_key_up = self._on_keyboard_up)
+        Window.bind(mouse_pos = self._on_mouse_pos)
+        Window.bind(on_close = self._on_close)
+        self.bind(size = self._on_size_changed)
+        
+        super(FigureCanvasKivy, self).__init__(figure, **kwargs)
+        _create_App(self)
+
+#     def draw(self):
+#         """
+#         Draw the figure using the KivyRenderer
+#         """
+
+        #renderer = RendererKivy(self.figure.dpi)
+        #self.figure.draw(renderer)
+
+    # You should provide a print_xxx function for every file format
+    # you can write.
+
+    # If the file type is not in the base set of filetypes,
+    # you should add it to the class-scope filetypes dictionary as follows:
+    filetypes = FigureCanvasBase.filetypes.copy()
+    filetypes['foo'] = 'My magic Foo format'
+    
+    def on_touch_down(self, touch):
+        if super(FigureCanvasKivy, self).on_touch_down(touch):
+            return True
+        if self.collide_point(*touch.pos):
```

Comment:
You need to generate `motion_notify_event`, `enter_notify_event`, and `leave_notify_event` events here. You generate this below for mouse, however, what if there's no mouse and just a touch screen, you still need to generate them. In fact, even with a mouse, you should generate it here, the only exception should be if the mouse is not held down, then you should generate them from `_on_mouse_pos`.